portal-bridge/src/bridge/history.rs

```diff
@@ -180,6 +194,7 @@ impl HistoryBridge {
             start: start_block,
             end: end_block,
         };
+        let gossip_send_semaphore = Arc::new(Semaphore::new(GOSSIP_LIMIT));
```

Comment:
I'd leave a comment here explaining why we're using a semaphore for backfill and the purpose of the `GOSSIP_LIMIT`

Reply:
small *nitpick* .. `to limit the amount of active gossip transfers to make sure we don't overwhelm the trin client`